### PR TITLE
Results Comp

### PR DESCRIPTION
**Pull request by flore77**
## Need

```gherkin
As a developer
I need a results component
So that I can display the test results
```

## Deliverables

A component that will render the results, i.e the Suites and the Tests.

## Solution

The `Report` component will render the `Results` component and will pass via props the whole data array.

The `Results` will iterate over and build an array with `Suite elements`, each element will receive as `key` the array `index`, i.e the position in the data array.

For now we will skip the first element, `data[0]`, i.e the rootSuite, because no one may have global tests and it is not worth to print it out.

Tests:

* should not render the root suite
* should render the suites array

Add one test for `Report comp`:
* should render the Results component

### CODE REVIEW

**Comment by flore77** [comment]
@andrei-picus-hs please review. sorry for the other commits...but I have created the branch from #9...my bad :disappointed:

**Comment by andrei-picus-hs** [comment]
@flore77 you can close this PR and open another one with the branch rebased onto master, or we can wait until #9 gets merged.

From http://git-scm.com/docs/git-rebase:

```sh
    o---o---o---o---o  master
         \
          o---o---o---o---o  next
                           \
                            o---o---o  topic

git rebase --onto master next topic

    o---o---o---o---o  master
        |            \
        |             o'--o'--o'  topic
         \
          o---o---o---o---o  next
```

**Comment by flore77** [comment]
Hmm. I will do a rebase, because I will learn something new :smile: . But a rebase on the feature-branch #8, I guess @andrei-picus-hs

**Comment by andrei-picus-hs** [comment]
Yep, I just copied the example from git-scm.

**Comment by flore77** [comment]
@andrei-picus-hs better we will wait for #9, because I need to fix some conflicts now, and than on merging I will need to fix some conflicts again...so better wait.

**Comment by flore77** [comment]
@andrei-picus-hs please review my code.

File: tests/unit/components/results.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  var tree;
+
+  beforeEach(function() {
+    tree = sd.shallowRender(<Results data={fixture}/>);
+  });
+
+  it('should not render the root suite', function() {
+    var suitesArray = tree.findNode('.results').props.children;
+
+    expect(suitesArray[0]).to.be.equal.null;
```
Comment (andrei-picus-hs) [inline tests/unit/components/results.js]: Ehm... if you want to assert that something is null you can either do `to.be.null` or `to.equal(null)`.

Also, quite an interesting check to see if something is not rendered. I'm guessing you're returning `null` from the `render` method if the suite is the root one.

You could also filter the suites first and remove the root one and then render them. That way you can check that either the length of `.results`' children is shorter by one, or that the root suite is not included in the list of the `suite` prop sent to the children.

```js
expect(tree.findNode('.results').props.children).to.have.length(fixture.length - 1);
// or
expect(_.map(tree.findNode('.results').props.children, 'props.suite')).to.not.include(fixture[0]);
```

I prefer the first type of check combined with the test below (should render the suites array) that includes the check that the root suite is not rendered.

File: tests/unit/components/results.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  });
+
+  it('should render the suites array', function() {
+    var suitesArray = tree.findNode('.results').props.children;
+
+    // Removes the root suite.
+    suitesArray.shift();
```
Comment (andrei-picus-hs) [inline tests/unit/components/results.js]: FYI: This mutates `props.children`. [slice(1)](http://devdocs.io/javascript/global_objects/array/slice) does not.

File: ui/components/results.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+var React = require('react'),
+    Suite = require('./suite.jsx');
+
+var Results = React.createClass({
+  render: function() {
+    var suites = this.props.data.map(function(suite, index) {
+      if (index === 0) {
```
Comment (andrei-picus-hs) [inline ui/components/results.jsx]: See [above](https://github.com/uberVU/mocha-mugshot-reporter/pull/12/files#r40656141).

File: ui/components/results.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    });
+
+    return <section className="results">
+      {suites}
+    </section>
+  }
+});
+
+module.exports = Results;
+
```
Comment (andrei-picus-hs) [inline ui/components/results.jsx]: Remove empty lines.

File: ui/components/report.jsx [hunk 5]
```diff
@@ -30,6 +31,7 @@ var Report = React.createClass({
     return <div className="report">
       <Header passes={passes.length} failures={failures.length}
         duration={duration}/>
+      <Results data={this.props.data}/>
```
Comment (andrei-picus-hs) [inline ui/components/report.jsx]: `suites` would be a better name for the prop and you also use that name [here](https://github.com/uberVU/mocha-mugshot-reporter/blob/414b3f58fe60511c62dbd5bf08ea351bda2bfcc6/ui/components/report.jsx#L11).

**Comment by andrei-picus-hs** [comment]
Finished review.

**Comment by flore77** [comment]
@andrei-picus-hs please review

**Comment by andrei-picus-hs** [comment]
LGTM!

**Comment by flore77** [comment]
:tada:  :tada:  maybe I will finish all the comp today :smile:

**Comment by andrei-picus-hs** [comment]
giphy

**Pull request merged**